driver/utils/type_info.h

```diff
@@ -0,0 +1,2112 @@
+#pragma once
+
+#include "driver/platform/platform.h"
+#include "driver/utils/unicode_conv.h"
+#include "driver/exception.h"
+
+#include <algorithm>
+#include <sstream>
+#include <string>
+#include <limits>
+#include <map>
+
+#include <cstring>
+
+#define lengthof(a) (sizeof(a) / sizeof(a[0]))
+
+struct TypeInfo {
+    std::string sql_type_name;
+    bool is_unsigned;
+    SQLSMALLINT sql_type;
+    int32_t column_size;
```

Comment:
Could you please add comments explaining what is the difference between `column_size` and `octet_length` ?

Reply:
These are typical ODBC concepts with lengthy descriptions provided here: https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/column-size-decimal-digits-transfer-octet-length-and-display-size

Reply:
Ok, please add this link as a comment

Reply:
ok